gpMgmt/bin/gppylib/programs/clsRecoverSegment.py

```
@@ -251,6 +251,9 @@ def run(self):
         if self.__options.differentialResynchronization and self.__options.outputSampleConfigFile:
             raise ProgramArgumentValidationException("Invalid -o provided with --differential argument")
 
+        if self.__options.disableReplayLag and not self.__options.rebalanceSegments:
+            raise ProgramArgumentValidationException("--disable-replay-lag should be used only with -r")
+
```

should also throw this exception for `--replay-lag`.

---

tried to add the above validation for --replay-lag as well but couldn't. Because we are assigning a default value of 10 GB to replay_lag so there is no way to know if user explicitly used --replay-lag or not. Even if --replay-lag is used with options other than -r it will not impact anything